core/domain/rights_manager.py

```diff
@@ -242,6 +248,28 @@ def is_exploration_cloned(exploration_id):
     return bool(exploration_rights.cloned_from)
 
 
+def _update_exploration_first_published_in_ms(
+    exploration_id, first_published_in_ms):
+    exploration_rights = get_exploration_rights(exploration_id)
+    exploration_rights.first_published_in_ms = first_published_in_ms
+    commit_cmds = [{
+        'cmd': CMD_UPDATE_FIRST_PUBLISHED,
+        'first_published': first_published_in_ms
+    }]
+    _save_activity_rights(feconf.SYSTEM_COMMITTER_ID, exploration_rights, ACTIVITY_TYPE_EXPLORATION,
+        'set first published time in ms', commit_cmds)
+
+
+# This function guards against the case that an exploration is published, then
+# unpublished, and then republished. The first_published time in ms should be the
+# time in ms of the first publication.
+def update_exploration_first_published_in_ms_if_not_set(
```

Comment:
Can we combine these and the collection methods into a single one (update_activity_first_published_msec_if_necessary) that takes an activity_type, activity_id, and a published_msec -- which does a check to see whether the existing publication date is None or is greater than the value being set, and if so, replaces it with the new value?